mllib-dal/src/main/native/service.cpp

```diff
@@ -237,4 +240,15 @@ NumericTablePtr homegenToSyclHomogen(NumericTablePtr ntHomogen) {
 
     return ntSycl;
 }
+
+HomogenTablePtr createHomogenTableWithArrayPtr(size_t pNumTabData,
+                                               size_t numRows, size_t numClos,
+                                               sycl::queue queue) {
+    double *htableArray = reinterpret_cast<double *>(pNumTabData);
+    auto data = sycl::malloc_shared<double>(numRows * numClos, queue);
+    queue.memcpy(data, htableArray, sizeof(double) * numRows * numClos).wait();
+    return std::make_shared<homogen_table>(
+        queue, data, numRows, numClos,
+        detail::make_default_delete<const double>(queue));    
+}
```

Comment:
I modified this by returning std::make_shared directly. could you help me to do clang-format?